charts/spire/charts/spiffe-oidc-discovery-provider/templates/configmap.yaml

```
@@ -2,14 +2,17 @@
 {{- define "spiffe-oidc-discovery-provider.yaml-config" -}}
 {{- $oidcSocket := .oidcSocket }}
 {{- with .root }}
+{{- $uri := urlParse (include "spire-lib.jwt-issuer" .) }}
 log_level: {{ .Values.config.logLevel | quote }}
 
 domains:
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}"
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ include "spiffe-oidc-discovery-provider.namespace" . }}"
   - "{{ include "spiffe-oidc-discovery-provider.fullname" . }}.{{ include "spiffe-oidc-discovery-provider.namespace" . }}.svc.{{ include "spire-lib.cluster-domain" . }}"
-  {{- $jwtIssuer := include "spire-lib.jwt-issuer" . }}
-  {{- uniq (concat (list $jwtIssuer) .Values.config.additionalDomains) | toYaml | nindent 2 }}
+  - "{{- default $uri.path $uri.host }}"
+  {{- with .Values.config.additionalDomains }}
+  {{- toYaml . | nindent 2 }}
+  {{- end }}
```

How about:
```suggestion
  {{- uniq (concat (list $uri.host) .Values.config.additionalDomains) | toYaml | nindent 2 }}
```

Then it still has the deduplication logic.

---

I don't believe this is a solution looking for a problem. If somebody has added their jwtIssuer manually, then it will have the same domain twice. Does that cause any issues?
Also, $uri.host is only used when a protocol is defined. A non-protocol string comes in as `$url.path`

---

🤷 I was asked to put the uniq in there when I added the code in my own pr. I'd expect the same pushback if its removed?